File: src/pocketcloud/event/impl/player/PlayerConnectEvent.php
<?php

namespace pocketcloud\event\impl\player;

use pocketcloud\player\CloudPlayer;
use pocketcloud\server\CloudServer;

class PlayerConnectEvent extends PlayerEvent {

    public function __construct(
        CloudPlayer $player,
        private readonly CloudServer $server
    ) {
        parent::__construct($player);
    }

    public function getServer(): CloudServer {
        return $this->server;
    }
}

File: cloud-core/src/main/java/de/pocketcloud/cloud/event/impl/player/PlayerConnectEvent.java
package de.pocketcloud.cloud.event.impl.player;

import de.pocketcloud.cloud.player.CloudPlayer;
import de.pocketcloud.cloud.server.CloudServer;
import lombok.Getter;

public final class PlayerConnectEvent extends PlayerEvent {

    @Getter
    private final CloudServer server;

    public PlayerConnectEvent(CloudPlayer player, CloudServer server) {
        super(player);
        this.server = server;
    }
}
